### PR TITLE
Add Docker support

### PR DESCRIPTION
**Pull request by josecols**
This PR includes a Dockerfile that extends the MeTA Docker image to install metapy. Users can use the Dockerfile or the Docker image built from it to run Python programs that require metapy.

### CODE REVIEW

**Comment by josecols** [comment]
@dmcguire81, we need to add the `DOCKERHUB_USERNAME` and `DOCKERHUB_TOKEN` workflow variables to enable DockerHub pushes.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    python-is-python3 \
+    python3-dev \
+    python3-pip
+
+WORKDIR /metapy
+RUN git submodule update --init --recursive
+RUN mkdir build
+
+WORKDIR /metapy
+RUN sed -i 's:add_subdirectory(deps/meta EXCLUDE_FROM_ALL):find_package(MeTA 3.0.2 REQUIRED):g' CMakeLists.txt
```
Comment (dmcguire81) [inline Dockerfile:14]: So this just builds the `pybind11` code and not the dependencies?

**Review by dmcguire81 — APPROVED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+COPY . /metapy
+
+RUN apt-get update && apt-get install -y \
+    python-is-python3 \
+    python3-dev \
+    python3-pip
+
+WORKDIR /metapy
+RUN git submodule update --init --recursive
```
Comment (dmcguire81) [inline Dockerfile]: Docker build fails here.

File: Dockerfile (hunk 1 above)
Comment (josecols) [reply Dockerfile:14]: Yes, MeTA and its dependencies are already built as libraries.

**Pull request merged**